### ISSUE TITLE
Engine does not start when built on windows on arm

### ISSUE THREAD

**Issue by Rick-T** [godotengine/godot#94253]
### Tested versions

- Reproducible in master (97b8ad1)
- Not reproducible in 4.2.2-stable (there's a different error)

### System information

Godot v4.2.2.stable - Windows 10.0.26100 - Vulkan (Forward+) - integrated Snapdragon(R) X Elite - X1E80100 - Qualcomm(R) Adreno(TM) GPU (Qualcomm Incorporated; 31.0.57.0) - Snapdragon(R) X 12-core X1E80100 @ 3.40 GHz (12 Threads)

### Issue description

I am running Windows 11 on arm64. I am trying to build Godot from source because the official builds only provide x86_64 binaries for windows. I am following the documentation and build Godot using `scons platform=windows production=yes arch=arm64`.

The compilation seems to work fine, however when I try to run Godot I get the following error message and the editor does not start:
```
Unable to initialze OpenGL video driver

Your video card drivers seem not to support the required OpenGL 3.3 version.

If possible, consider updating you video card drivers.

If you have recently updated your video card drivers, try rebooting.
```

My video card drivers are up to date. OpenGL is working in other applications (e.g. Blender, which I have also built from source).

When I run the official 4.2.2 windows binary (x86_64) the engine also starts fine.

Does anybody have an idea what I might be doing wrong?

### Steps to reproduce

`git clone ...`
`scons platform=windows arch=arm64`
run godot.windows.editor.arm64.exe

### Minimal reproduction project (MRP)

[empty.zip](https://github.com/user-attachments/files/16191661/empty.zip)

**Comment by bruvzg** [comment]
There's no OpenGL drivers on most ARM Windows systems, you should build with DX12 (for mobile/forward+) and/or with ANGLE (for compatibility renderer), see https://docs.godotengine.org/en/latest/contributing/development/compiling/compiling_for_windows.html. Currently, there are no official ARM releases of the static libs, so you'll need to build them manually as well.

**Comment by Rick-T** [comment]
Thanks. I have now tried building with DX12. I have built godot-nir-static myself. I am not using ANGLE.

I have built godot-nir-static using the commands from the documentation:

- `git clone ...`
- `git submodule update --init`
- `./update_mesa.sh`
- `scons`

For the last step (`scons`) I had to add some imports to `godot-tools/targets.py` and `godot-tools/windows.py`. I am not sure why the files didn't work out of the box (maybe I have a wrong scons version?). However, I don't think my changes should affect the outcome.

I have added the following lines to `godot-tools/targets.py`:
```
from SCons.Variables.BoolVariable import BoolVariable
from SCons.Variables.EnumVariable import EnumVariable
```

I have added the following lines to `godot-tools/windows.py`:
```
import os
from SCons.Variables.BoolVariable import BoolVariable
```

I am now building Godot using `scons platform=windows d3d12=yes mesa_libs="<path>/godot-nir-static" production=yes`.

Compilation works fine, however I am not getting an error in the linking phase:

```
core.windows.editor.arm64.lib(engine.windows.editor.arm64.obj) : error LNK2001: unresolved external symbol "unsigned __int64 const VERSION_TIMESTAMP" (?VERSION_TIMESTAMP@@3_KB)
main.windows.editor.arm64.lib(main.windows.editor.arm64.obj) : error LNK2001: unresolved external symbol "unsigned __int64 const VERSION_TIMESTAMP" (?VERSION_TIMESTAMP@@3_KB)
editor.windows.editor.arm64.lib(project_manager.windows.editor.arm64.obj) : error LNK2001: unresolved external symbol "unsigned __int64 const VERSION_TIMESTAMP" (?VERSION_TIMESTAMP@@3_KB)
editor.windows.editor.arm64.lib(editor_bottom_panel.windows.editor.arm64.obj) : error LNK2001: unresolved external symbol "unsigned __int64 const VERSION_TIMESTAMP" (?VERSION_TIMESTAMP@@3_KB)
editor.windows.editor.arm64.lib(editor_about.windows.editor.arm64.obj) : error LNK2001: unresolved external symbol "unsigned __int64 const VERSION_TIMESTAMP" (?VERSION_TIMESTAMP@@3_KB)
bin\godot.windows.editor.arm64.exe : fatal error LNK1120: 1 unresolved externals
scons: *** [bin\godot.windows.editor.arm64.exe] Error 1120
scons: building terminated because of errors.
```

Any idea what went wrong?

**Comment by bruvzg** [comment]
`VERSION_TIMESTAMP` is generated header from the main repo, it's not related to the DX12. No idea why it's missing, try cleaning the build tree and making clean build, probably something was not generated. All Godot related repos should require SCons 4.3+.

**Comment by Rick-T** [comment]
Cleaning everything and building from scratch made it work.

After rebuilding I got the same OpenGL error as before. Then I realized that I can force Godot to use another rendering-driver. I am now running Godot with the `--rendering-driver vulkan` option, which seems to work fine. DX12 doesn't seem to be necessary.

Now I have the problem that many buttons in the editor don't respond to touchscreen events. But I guess that's an unrelated problem, so I'll close this issue.

Thanks again.  :heart:
